<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1869</width>
    <height>1383</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Raster Packer Evolution 3000 (v 0.1)</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Item:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="spinBox">
          <property name="enabled">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="comboBox">
          <property name="enabled">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>X:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="doubleSpinBox">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="maximum">
           <double>99999.990000000005239</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Y:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="doubleSpinBox_2">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="maximum">
           <double>99999.990000000005239</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Orientation:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="comboBox_2">
          <property name="enabled">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="Line" name="line_5">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="comboBox_3">
          <property name="minimumSize">
           <size>
            <width>170</width>
            <height>0</height>
           </size>
          </property>
          <item>
           <property name="text">
            <string>Configure...</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Default (no metaheuristic)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Default (Guided Local Search)</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="resources.qrc">
            <normaloff>:/img/PlayerPlay.png</normaloff>:/img/PlayerPlay.png</iconset>
          </property>
          <property name="flat">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_2">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="resources.qrc">
            <normaloff>:/img/PlayerStop.png</normaloff>:/img/PlayerStop.png</iconset>
          </property>
          <property name="flat">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <widget class="Line" name="line_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>Initial Solution</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_5">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Create Random Solution</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_32">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Bottom Left</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_14">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Set Container Width</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_19">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Set Container Height</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="Line" name="line_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>No metaheuristic</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_3">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="minimumSize">
             <size>
              <width>150</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string>Display Total Overlap Map</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_4">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Translate To Minimum</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_6">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Global Overlap</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_7">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Local Search</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="Line" name="line_4">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Guided Local Search</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_10">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="minimumSize">
             <size>
              <width>150</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string>Display Total Overlap Map</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_9">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Translate To Minimum</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_8">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Local Search</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_12">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Update Weights</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_13">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Reset Weights</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="Line" name="line_6">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>Zooming Approach</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_18">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Display Search Overlap Map</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_21">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Display Search Overlap Map Single</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_16">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Display Zoomed Map</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_17">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Translate to Minimum</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_20">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Local Search</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_22">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Set Search Map Scale</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="Line" name="line_7">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>Item Clustering</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_11">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Switch to Unclustered Problem</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="Line" name="line_8">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_15">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Print Solution</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="Line" name="line">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="PackingViewer" name="graphicsView">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>350</width>
            <height>0</height>
           </size>
          </property>
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1869</width>
     <height>31</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLoad_Problem"/>
    <addaction name="separator"/>
    <addaction name="actionLoad_Solution"/>
    <addaction name="actionSave_Solution"/>
    <addaction name="separator"/>
    <addaction name="actionExport_Solution_to_SVG"/>
    <addaction name="actionExport_Solution_to_Tikz"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
    <addaction name="actionShow_density"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuTools"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionLoad_Problem">
   <property name="text">
    <string>Load Problem</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionShow_Items">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show Items</string>
   </property>
  </action>
  <action name="actionShow_Map">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show Map</string>
   </property>
  </action>
  <action name="actionLoad_Zoomed_Problem">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Load Zoomed Problem</string>
   </property>
  </action>
  <action name="actionLoad_Solution">
   <property name="text">
    <string>Load Solution</string>
   </property>
  </action>
  <action name="actionSave_Solution">
   <property name="text">
    <string>Save Solution</string>
   </property>
  </action>
  <action name="actionExport_Solution_to_SVG">
   <property name="text">
    <string>Export Solution to SVG</string>
   </property>
  </action>
  <action name="actionGet_density">
   <property name="text">
    <string>Show density</string>
   </property>
  </action>
  <action name="actionShow_density">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Show density</string>
   </property>
  </action>
  <action name="actionSave_Zoomed_Solution">
   <property name="text">
    <string>Save Zoomed Solution</string>
   </property>
  </action>
  <action name="actionExport_Solution_to_Tikz">
   <property name="text">
    <string>Export Solution to Tikz</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>PackingViewer</class>
   <extends>QGraphicsView</extends>
   <header>viewer/packingviewer.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections/>
</ui>
